cmd/aws-k8s-agent/main.go

```diff
@@ -39,18 +37,17 @@ func _main() int {
 
 	log.Infof("Starting L-IPAMD %s  ...", version)
 
-	kubeClient, err := k8sapi.CreateKubeClient()
-	if err != nil {
-		log.Errorf("Failed to create client: %v", err)
+	//Check API Server Connectivity
+	if k8sapi.CheckAPIServerConnectivity() != nil{
 		return 1
 	}
 
-	eniConfigController := eniconfig.NewENIConfigController()
-	if ipamd.UseCustomNetworkCfg() {
-		go eniConfigController.Start()
+	standaloneK8SClient, k8sClient, err := k8sapi.CreateKubeClients()
+	if err != nil{
+		return 1
 	}
 
-	ipamContext, err := ipamd.New(kubeClient, eniConfigController)
+	ipamContext, err := ipamd.New(standaloneK8SClient, k8sClient)
```

Comment:
Why does ipamd.New need both cached and uncached client?  That seems weird, or a sign that we need cache invalidation somewhere.